pkg/util/test/util.go

```diff
@@ -3083,24 +3062,55 @@ func ValidateTelemetryV2Enabled(pxImageList map[string]string, cluster *corev1.S
 	return nil
 }
 
-func validateTelemetryHealthyInPxctl(namespace string) bool {
+func validateTelemetryStatusInPxctl(telemetryShouldBeEnabled bool, namespace string) error {
+	listOptions := map[string]string{"name": "portworx"}
 	cmd := "pxctl status | grep Telemetry:"
-	ignoreErr := false
 
-	output, err := runCmdInsidePxPod(cmd, namespace, ignoreErr)
-	if err == nil {
-		if strings.Contains(output, "Healthy") {
-			logrus.Infof("Telemetry status in pxctl [%v]", strings.TrimSpace(output))
-			return true
+	logrus.Infof("Validate Telemetry pxctl status on each PX node")
+	t := func() (interface{}, bool, error) {
+		// Get Portworx pods
+		pxPods, err := coreops.Instance().GetPods(namespace, listOptions)
+		if err != nil {
+			return nil, true, fmt.Errorf("failed to get PX pods, Err: %v", err)
 		}
 
-		// If Telemetry is not Healthy, we are assuming any other status is Disabled and/or Unhealthy
-		logrus.Warnf("Telemetry status in pxctl [%v]", strings.TrimSpace(output))
-		return false
+		podsWithTelemetryStatusOk := 0
+		for _, pxPod := range pxPods.Items {
+			// Validate PX pod is ready to run command on
+			if !coreops.Instance().IsPodReady(pxPod) {
+				return nil, true, fmt.Errorf("[%s] PX pod is not in Ready state", pxPod.Name)
+			}
+			logrus.Debugf("PX pod is ready to execute command on [%s]", pxPod.Name)
+
+			output, err := runCmdInsidePxPod(&pxPod, cmd, namespace, false)
+			if err != nil {
+				return nil, true, fmt.Errorf("got error while trying to get Telemetry status from pxctl, Err: %v", err)
+			}
+
+			if strings.Contains(output, "Healthy") && telemetryShouldBeEnabled == true {
+				logrus.Infof("[%s] Telemetry status in pxctl [%v]", pxPod.Name, strings.TrimSpace(output))
```

Comment:
2 cases here
1. if telemetry enabled, and result doesn't contain Healthy, return to retry directly
2. if telemetry disabled, and result doesn't contain Disabled, return to retry directly


this should make it simpler. we already scan all px pods, don't need to count the results

Reply:
Done